common/app/views/fragments/analytics/google.scala.html

```diff
@@ -59,6 +61,7 @@
     }
     var isLoggedIn = (!!identityId).toString();
 
+    delete window.ga;
```

Comment:
Looking at this now, I'd quite like to check that we are still seeing late events like attention time tracking with this line

Reply:
What's the purpose of deleting it in the initalisation?

Reply:
We define a stub function on [line 18](https://github.com/guardian/frontend/pull/23203/files#diff-55b65cfe67f24b5a3598c9d4bbd50fb8b8e4ad5dd5d2bdbf557ee66a61875201R18) `window.ga = function() {};` - this is so that code calling window.ga doesn't error before we load/if we don't load the tracking script because of consents.

We delete it just before it is initialised because it doesn't get initialised if the stub already exists.

Looking at it again, I think because the stub is being deleted and the proper script initialised, this should be fine.

This is all a horrible hack.

Reply:
I see, it'd be great to add some guidance on that (or even a link to this comment that explains it well!)